django_react_proj/users-fe/src/components/SummingTime/SummingTime.js

```diff
@@ -1,59 +1,55 @@
 import { Pie } from '@ant-design/charts';
-import { Col, Row } from 'reactstrap';
-import React, { useEffect, useState } from 'react';
-import axios from 'axios';
-import { API_URL } from '../../constants';
+import { Card, Row, Col } from 'antd';
+import React from 'react';
+import PropTypes from 'prop-types';
 import styles from './SummingTime.module.css';
 
-const Subtract = () => {
-  const [users, setUsers] = useState();
+const Subtract = ({ users }) => {
   let timevalue = 0;
-  const getUsers = () => {
-    axios.get(API_URL).then((res) => setUsers(res.data));
-  };
-
-  useEffect(() => {
-    const handle = setInterval(getUsers, 1000);
-    return () => clearInterval(handle);
-  }, []);
+  let done = 0;
+  let notdone = 0;
 
   if (users) {
-    users.forEach((user) => {
+    Array.prototype.forEach.call(users, (user) => {
       const uptimes = new Date(user.endTime).getTime();
       const lowtimes = new Date(user.startTime).getTime();
       const sub = uptimes - lowtimes;
       const minutes = sub / 60000;
       timevalue += minutes;
     });
   }
+  if (users) {
+    Array.prototype.forEach.call(users, (user) => {
+      const itisdone = String(user.done);
+      if (itisdone === 'true') {
+        done += 1;
+      } else {
+        notdone += 1;
+      }
+    });
+  }
   const taskhours = Math.floor(timevalue / 60);
   const taskminutes = timevalue % 60;
   const substrpiechart = 480 - timevalue;
   const datatask = [
     {
       type: 'Task Done',
-      value: 3,
+      value: done,
     },
     {
       type: 'Task Not Done',
-      value: 4,
+      value: notdone,
     },
   ];
   const config = {
```

Comment:
Maybe extract configured Pie to separate component?

Reply:
Done